umqtt.simple/umqtt/simple.py

```diff
@@ -53,24 +54,45 @@ def set_last_will(self, topic, msg, retain=False, qos=0):
 
     def connect(self, clean_session=True):
         self.sock = socket.socket()
+        self.addr = socket.getaddrinfo(self.server, self.port)[0][-1]
```

Comment:
I'm afraid these and above changes are unrelated to the topic of this PR and should be submitted separately.